src/builder/rollup.ts

```diff
@@ -140,6 +140,9 @@ export function getRollupOptions (ctx: BuildContext): RollupOptions {
         return false
       }
       if (!isExplicitExternal) {
+        if (ctx.options.failOnWarn) {
+          throw new Error(`Implicit external ${id} detected. Either add to your project dependencies or add to \`externals\`. You can remove this protection by disabling \`failOnWarn\`.`)
```

Comment:
What if we add a _warnings_ field to the context and lately in the build process check on them and fail with code (1) if there were any?